src/main/java/org/opentripplanner/middleware/trip_monitor/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,176 @@
+                break;
+            case "email":
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            case "all":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void checkTripForNewAlerts(MonitoredTrip trip, Itinerary itinerary) {
+        if (!trip.notifyOnAlert) {
+            LOG.debug("Notify on alert is disabled for trip {}. Skipping check.", trip.id);
+            return;
+        }
+        // FIXME: Need to somehow determine how to get last itinerary from responses in journey state.
+        Itinerary lastItinerary = trip.lastItinerary();
+        List<LocalizedAlert> newAlerts = itinerary.getAlerts();
+        List<LocalizedAlert> previousAlerts = lastItinerary.getAlerts();
+        // If journey state is already tracking alerts from previous checks, see if they have changed.
+        if (previousAlerts.size() > 0) {
+            // If there are new alerts, notify user.
+            List<LocalizedAlert> unseenAlerts = new ArrayList<>();
+            for (LocalizedAlert newAlert : newAlerts) {
+                boolean alertIsNew = true;
+                for (LocalizedAlert previousAlert : previousAlerts) {
+                    if (newAlert.equals(previousAlert)) {
+                        alertIsNew = false;
+                        break;
+                    }
+                }
+                if (alertIsNew) unseenAlerts.add(newAlert);
+            }
+            if (unseenAlerts.size() > 0) {
+                // FIXME: May need to update the user about alerts that have dropped off as well.
+                enqueueNotification(TripMonitorNotification.createAlertNotification(unseenAlerts));
+            }
+        } else {
+            // If there are new alerts present in the response, create notification.
+            if (newAlerts.size() > 0) {
+                enqueueNotification(TripMonitorNotification.createAlertNotification(newAlerts));
+            }
+        }
+    }
+
+    private void enqueueNotification(TripMonitorNotification tripMonitorNotification) {
+        notifications.add(tripMonitorNotification);
+    }
+
+    private boolean shouldSkipMonitoredTripCheck(MonitoredTrip trip) {
+        LocalDate now = LocalDate.now();
+        LocalDate tripTime;
+        try {
+            tripTime = DateUtils.getDateFromString(trip.tripTime, "HH:mm");
+        } catch (DateTimeParseException e) {
+            // TODO: Bugsnag report
+            LOG.error("Could not parse trip time", e);
+            return false;
+        }
+        // Trip is not active today.
+        return !trip.isActiveOnDate(now) &&
+            // Trip has already occurred.
+            tripTime.isAfter(now);
```

Comment:
This will return true for the whole day. So trips that happen at like 10pm will be checked throughout the day. There should probably be some more checks in here.